flow/Net2.actor.cpp

```diff
@@ -790,6 +800,17 @@ struct SSLHandshakerThread final : IThreadPoolReceiver {
 		Handshake(ssl_socket& socket, ssl_socket::handshake_type type) : socket(socket), type(type) {}
 		double getTimeEstimate() const override { return 0.001; }
 
+		std::string getPeerIP() const {
+			try {
+				return socket.lowest_layer().remote_endpoint().address().to_string();
+			} catch (boost::system::system_error e) {
```

Comment:
I'm all for exception safety but I don't think there's any parsing involved here.
It's stringifying a binary address format.
Is this exception really possible?

Reply:
remote_endpoint() can throw if the other end of the connection has disconnected when remote_endpoint() is called. However, we can also just pass an `boost::system::error_code`  to the function and get a non-throwing version. Seeing this I think this is the better option.